### PR TITLE
[gbc c++] Add missing move assignment operator

### PR DESCRIPTION
**Pull request by ara-ayvazyan**

### CODE REVIEW

File: compiler/src/Language/Bond/Codegen/Cpp/Types_h.hs [hunk 1]
```diff
@@ -309,12 +309,13 @@ namespace std
             -- default OK when there are no meta fields
             implicitlyDeclared = CPP.ifndef CPP.defaultedFunctions [lt|
         // Compiler generated operator= OK
-        #{declName}& operator=(const #{declName}&) = default;|]
+        #{declName}& operator=(const #{declName}&) = default;
+        #{declName}& operator=(#{declName}&&) = default;|]
```
Comment (ara-ayvazyan) [inline compiler/src/Language/Bond/Codegen/Cpp/Types_h.hs]: Looking at VC12 compatibility.

File: compiler/tests/generated/alias_key_types.h [hunk 2]
```diff
@@ -45,9 +45,16 @@ namespace test
 #endif
         
         
-#ifndef BOND_NO_CXX11_DEFAULTED_FUNCTIONS
+#if !defined(BOND_NO_CXX11_DEFAULTED_MOVE_CTOR)
         // Compiler generated operator= OK
         foo& operator=(const foo&) = default;
+        foo& operator=(foo&&) = default;
+#else
+        foo& operator=(foo other)
```
Comment (chwarr) [inline compiler/tests/generated/alias_key_types.h]: What is the result of `std::is_(nothrow_)move_assignable` for a type that just has this kind of assignment operator?

Comment (ara-ayvazyan) [reply compiler/tests/generated/alias_key_types.h]: The `std::is_move_assignable` will of course be `true`, and the `std::is_nothrow_move_assignable` will be `false`. Note, that STL containers gained `noexcept` only in C++17 for their move-assignment operators.

**Comment by lalo** [comment]
@ara-ayvazyan do we have tests that exercise and compile this c++ code?

**Pull request merged**